File: java-concert/src/main/java/org/example/javaconcert/concert/presentation/response/ConcertResponse.java
package org.example.javaconcert.concert.presentation.response;

import java.time.LocalDate;

import org.example.javaconcert.concert.domain.Concert;
import org.example.javaconcert.concert.domain.Tag;

public record ConcertResponse(
	String title,
	String location,
	LocalDate startAt,
	LocalDate endAt,
	Tag tag
) {

	public static ConcertResponse toResponse(Concert concert) {
		return new ConcertResponse(concert.getTitle(),
			concert.getLocation(),
			concert.getStartAt(),
			concert.getEndAt(),
			concert.getTag());
	}
}

File: concert/src/main/kotlin/prac/concert/concert/presentation/response/ConcertResponse.kt
package prac.concert.concert.presentation.response

import prac.concert.concert.domain.Concert
import prac.concert.concert.domain.Tag
import java.time.LocalDate

data class ConcertResponse(
    val title: String,
    val location: String,
    val startAt: LocalDate,
    val endAt: LocalDate,
    val tag: Tag
) {

    companion object{
        fun toResponse(concert: Concert): ConcertResponse {
            return ConcertResponse(
                title = concert.title,
                location = concert.location,
                startAt = concert.startAt,
                endAt = concert.endAt,
                tag = concert.tag
            )
        }
    }

}
